vignettes/datatable-reference-semantics.Rmd

```diff
@@ -112,6 +112,8 @@ Note that the code above explains how `:=` can be used. They are not working exa
 
 * On the other hand, (b) is handy if you would like to jot some comments down for later.
 
+* In general, (a) and (b) should produce the same result, however there are minor differences in certain cases due to the implementation details of `:=`. Specifically, when using (b), the `:=` operator behaves like an alias to `list`, therefore if `RHS` is a `list` in functional form the column will become a list of lists.
```

Comment:
thanks, that is better, but a code example would be even more helpful, is that possible?

Reply:
Will do!

Reply:
LMK if example makes sense